tests/models/test_offer.py

```diff
@@ -133,3 +132,12 @@ def test_get_offer_files_with_creative_code(offer):
     assert isinstance(files[0], OfferFile)
     assert files[0].offer_id == offer.id
     assert files[0].creativecode == 'http://www.example.com'
+
+
+def test_set_affiliate_approval_status(api):
+    assert api.offers.set_affiliate_approval_status(472, 20, 'approved') is True
+
+
+def test_offer_set_affiliate_approval_status(offer):
+    assert isinstance(offer.set_affiliate_approval_status(20, 'approved'), AffiliateOffer)
+    assert isinstance(offer.set_affiliate_approval_status(20, 'approved'), bool)
```

Comment:
Is it an error? Does HO really return object on the first call and boolean on the next one?

Reply:
No, it's not. It returns an object in case one was created, and just a boolean, instead if existing object was updated\failed to be updated. https://developers.tune.com/network/offer-setaffiliateapproval/ :

    Variant: if new AffiliateOffer object is created, that object's ID.
    Otherwise, true if update succeeds; false if not.